Hw06 email

Marta7799 opened the pull request.

Marta7799 committed:
installed  sendgrid/mail

Marta7799 committed:
step 1,2,3,4

Marta7799 committed:
updated package.json

changed nanoid 5.0 to 3.0 because of errors

KamilMr commented on controllers/auth/signup.js:
```diff
@@ -13,8 +15,24 @@ const signup = async (req, res) => {
+  const mail = {
+    to: email,
+    subject: "Approve email",
+    html: `<a a target="_blank" hreaf="http://localhost:3000/api/users/verify/${verificationToken}">Approve email</a>`,
```
Czy przenosił Cię do tego linku? Bo jest literówka, powinien być `href` atrybut

Marta7799 replied:
Racja :)

KamilMr commented on helpers/sendEmail.js:
```diff
@@ -0,0 +1,19 @@
+  try {
+    await sgMail.send(email);
+    return true;
+  } catch (error) {
```
```suggestion
  } catch (error) {
  console.log(error)
```
dobrze jeszcze mieć console log. W razie błędu

Marta7799 replied:
ok

KamilMr commented on models/user.js:
```diff
@@ -43,10 +51,15 @@ const joiLoginSchema = Joi.object({
 });
 
+const verifyEmailSchema = Joi.object({
+  email: Joi.string().required(),
```
To trzeba bardziej zacieśnić, Joi sprawdza też wiadomości email.

Marta7799 replied:
Czyli np dodać email(). ?

KamilMr commented on routes/api/users.js:
```diff
@@ -1,17 +1,23 @@
   upload.single("avatar"),
   ctrlWrapper(ctrl.updateAvatar)
 );
+router.get("/verify/:verificationToken", ctrlWrapper(ctrl.verifyEmail));
+router.post(
+  "/verify",
+  validation(verifyEmailSchema),
+  ctrlWrapper(ctrl.resendVerifyEmail)
+);
```
Nie widzę nigdzie dodanej logiki która obsługuje weryfikacje tokenu oraz ponownego wysłania email.

Marta7799 replied:
Dodane :)

KamilMr reviewed (changes requested):
Hej, 
Nie widzę nigdzie dodanej logiki która obsługuje weryfikacje tokenu oraz ponownego wysłania email, dodaj ja jeszcze pls. :)

Marta7799 committed:
updated controllers,

Poprawiona literówka w href, dodane ponowne wysłanie emaila, dodany console log do błedu,, dodanie sprawdzenie emaila

Marta7799 committed:
added verifyEmail

KamilMr reviewed (approved):
Hejka zadanie zatwierdzam :)
